tests/BootstrapperTest.php

```diff
@@ -380,3 +381,30 @@ function getDiskPrefix(string $disk): string
 
     return $prefix;
 }
+
+test('url bootstrapper overrides the root url when tenancy gets initialized and reverts the url to the central one after tenancy ends', function() {
+    config(['tenancy.bootstrappers.url' => UrlTenancyBootstrapper::class]);
+
+    UrlTenancyBootstrapper::$rootUrlOverride = function (Tenant $tenant) {
+        $baseUrl = URL::to('/');
+        $scheme = str($baseUrl)->before('://') . '://';
+
+        return str($baseUrl)
+            ->after($scheme)
+            ->prepend($tenant->getTenantKey() . '.')
+            ->prepend($scheme)
```

Comment:
This syntax is a bit confusing, so for the docs we may want to think of something else (e.g. `replaceFirst($scheme, $scheme . "://$subdomain.")` or something like that)

Reply:
Actually it's not just in the docs, it's also in the code example. So try to rewrite it into something that's intuitively understandable and then we can merge the PR. The code looks good to me.

Reply:
Updated this https://github.com/archtechx/tenancy/pull/1044/commits/3fbf6b4a45932a0d4ccbae3c363d2198fa9e4648#diff-b22931f4c76926f01513d299f2f87fb1528e0d3a92e539ef66cb4675b4342f4bR401-R406

Reply:
Should be updated in the comment example as well.

Reply:
And you can probably keep all of it inside the closure, that would be cleaner. Just avoid using `after()->prepend()->prepend()`

Reply:
[I moved the variables to the closure](https://github.com/archtechx/tenancy/pull/1044/commits/7bcf9d329ac91d83d0f01ce4a157b9f2dd8f8f72) (except for `$baseUrl`), and [updated the comment example](https://github.com/archtechx/tenancy/pull/1044/commits/5e78d03ea40fa05310863bedadb20a8774ff18a6)